x-pack/plugins/fleet/server/services/agents/reassign.ts

```diff
@@ -35,6 +37,25 @@ export async function reassignAgent(
   });
 }
 
+export async function reassignAgentIsAllowed(
+  soClient: SavedObjectsClientContract,
+  esClient: ElasticsearchClient,
+  agentId: string,
+  newAgentPolicyId: string
+) {
+  const agentPolicy = await getAgentPolicyForAgent(soClient, esClient, agentId);
+  if (agentPolicy?.is_managed) {
+    throw new Error(`Cannot reassign an agent from managed agent policy ${agentPolicy.id}`);
```

Comment:
Should we create a specific Fleet error for that? it will allow us to handle this in the error handler and return a 400 as it is a user error  more than a server error the user try to do a forbidden action on a policy

Reply:
Totally. Thanks for reminding me. Added in https://github.com/elastic/kibana/pull/88688/commits/86d69589d618c9cb8ddcfff22489c2d5ec9aab37